### ISSUE TITLE
mbed-test-wrapper cause failure

### ISSUE THREAD

**Issue by LiyouZhou** [autopulated/mbed-test-wrapper#2]
Is this still the supported way of running tests?

If use "yt test", test fails with error:

```
/work/iot/minar$ yt test
info: generate for target: frdm-k64f-gcc 2.0.0 at /work/iot/minar/yotta_targets/frdm-k64f-gcc
GCC version is: 4.9.3
-- Configuring done
-- Generating done
-- Build files have been written to: /work/iot/minar/build/frdm-k64f-gcc
[1/2] Building CXX object test/CMakeFiles/minar-test-dispatchtest.dir/work/iot/minar/test/dispatchtest.cpp.o
In file included from /work/iot/minar/test/dispatchtest.cpp:23:0:
/work/iot/minar/yotta_modules/mbed-drivers/mbed/mbed.h:19:2: warning: #warning mbed/mbed.h is deprecated. Please use mbed-drivers/mbed.h instead. [-Wcpp]
 #warning mbed/mbed.h is deprecated.  Please use mbed-drivers/mbed.h instead.
  ^
In file included from /work/iot/minar/test/dispatchtest.cpp:26:0:
/work/iot/minar/yotta_modules/mbed-drivers/mbed/test_env.h:19:2: warning: #warning mbed/test_env.h is deprecated. Please use mbed-drivers/test_env.h instead. [-Wcpp]
 #warning mbed/test_env.h is deprecated.  Please use mbed-drivers/test_env.h instead.
  ^
[2/2] Linking CXX executable test/minar-test-dispatchtest
info: using filter "mbedgt --digest stdin -v -V" for tests in /work/iot/minar/build/frdm-k64f-gcc/test
info: test minar: minar-test-complex_dispatch
mbedgt: selecting test case observer...
    selected digest source: stdin
MBED: Instrumentation: "/dev/ttyACM0" and disk: "/media/MBED"
HOST: Copy image onto target...
HOST: Initialize serial port...
...port ready!
HOST: Reset target...
setting duration: 10
HOST: Detecting test case properties...
HOST: Error! No host test name defined in preamble

{{error}}
{{end}}
Hmbedgt: mbed-host-test-runner: stopped
mbedgt: mbed-host-test-runner: returned 'ERROR'
completed in 10.44 sec
error: test minar-test-complex_dispatch failed (command: minar-test-complex_dispatch)
info: test minar: minar-test-dispatchtest
mbedgt: selecting test case observer...
    selected digest source: stdin
MBED: Instrumentation: "/dev/ttyACM0" and disk: "/media/MBED"
HOST: Copy image onto target...
HOST: Initialize serial port...
...port ready!
HOST: Reset target...
setting duration: 10
HOST: Detecting test case properties...
HOST: Error! No host test name defined in preamble

{{error}}
{{end}}
Hmbedgt: mbed-host-test-runner: stopped
mbedgt: mbed-host-test-runner: returned 'ERROR'
completed in 10.31 sec
error: test minar-test-dispatchtest failed (command: minar-test-dispatchtest)
info: tests complete: 0 passed, 2 failed
```

If I use `mbedgt` insteat, it all works fine.

```
/work/iot/minar$  mbedgt -t frdm-k64f-gcc
mbedgt: detecting connected mbed-enabled devices...
mbedgt: detected 1 device
    detected 'K64F' -> 'K64F[0]', console at '/dev/ttyACM0', mounted at '/media/MBED', target id '024002261e501e62000000000000000000000000e389e3da'
mbedgt: local yotta target search in './yotta_targets' for compatible mbed-target 'k64f'
    inside './yotta_targets/frdm-k64f-gcc' found compatible target 'frdm-k64f-gcc'
mbedgt: processing 'frdm-k64f-gcc' yotta target compatible platforms...
mbedgt: processing 'K64F' platform...
mbedgt: using platform 'K64F' for test:
    target_id_mbed_htm = '024002261e501e62000000000000000000000000e389e3da'
    mount_point = '/media/MBED'
    target_id = '024002261e501e62000000000000000000000000e389e3da'
    serial_port = '/dev/ttyACM0'
    target_id_usb_id = '024002261e501e62000000000000000000000000e389e3da'
    platform_name = 'K64F'
    platform_name_unique = 'K64F[0]'
mbedgt: building your sources and tests with yotta...
    calling yotta: yotta --target=frdm-k64f-gcc,* build
info: generate for target: frdm-k64f-gcc 2.0.0 at /work/iot/minar/yotta_targets/frdm-k64f-gcc
GCC version is: 4.9.3
-- Configuring done
-- Generating done
-- Build files have been written to: /work/iot/minar/build/frdm-k64f-gcc
ninja: no work to do.
mbedgt: yotta build for target 'frdm-k64f-gcc' was successful
mbedgt: running 2 tests for target 'frdm-k64f-gcc' and platform 'K64F'
    use 1 instance for testing
setting duration: 10
setting duration auto timeout 35
mbedgt: test on hardware with target id: 024002261e501e62000000000000000000000000e389e3da 
    test 'minar-test-complex_dispatch' ..................................................... OK in 31.28 sec
setting duration: 10
setting duration auto timeout 20
mbedgt: test on hardware with target id: 024002261e501e62000000000000000000000000e389e3da 
    test 'minar-test-dispatchtest' ......................................................... OK in 16.24 sec
mbedgt: test report:
+---------------+---------------+-----------------------------+--------+--------------------+-------------+
| target        | platform_name | test                        | result | elapsed_time (sec) | copy_method |
+---------------+---------------+-----------------------------+--------+--------------------+-------------+
| frdm-k64f-gcc | K64F          | minar-test-complex_dispatch | OK     | 31.28              | shell       |
| frdm-k64f-gcc | K64F          | minar-test-dispatchtest     | OK     | 16.24              | shell       |
+---------------+---------------+-----------------------------+--------+--------------------+-------------+

Result: 2 OK
completed in 63.76 sec
```

**Comment by autopulated** [comment]
Sorry I wasn't watching this repo – yes, this should work. Possibly the mbedhtrun / greentea / ls APIs that mbed-test-wrapper uses have changed, are you still experiencing this problem?

**Comment by autopulated** [comment]
It turns out this is due to a baud-rate mismatch. htrun now expects the tty device to have the speed munged into it (`/dev/tty.whatever:9600`, for example).

**Comment by autopulated** [comment]
Published version 1.0.0 which works around this problem.
